### ISSUE TITLE
Handling access concurrency

### ISSUE THREAD

**Issue by giovanni-orciuolo** [bartventer/gorm-multitenancy#47]
Hello! I was wondering how access concurrency is handled inside this package. I will explain the scenario: 

Assuming a setup with a single PostgreSQL database with one schema per tenant, let's say that 2 users connect at the same time with different tenants. Since the tenant is set at database connection level, and that's shared between HTTP requests, there could be a situation where the first user sees stuff from the second user tenant since the schema was switched before performing the query.

How is this case handled?

**Comment by bartventer** [comment]
Thank you for your question about access concurrency. I understand the concern about schema switching when multiple users connect with different tenants simultaneously.

For PostgreSQL, we use transaction-level advisory locks for tenant migrations to ensure isolation. When changing the search path, we recommend doing so within a transaction or using a separate database connection for each tenant, if practical. This maintains data isolation and prevents cross-tenant data visibility. We recently added a convenience method [DB.WithTenant](https://pkg.go.dev/github.com/bartventer/gorm-multitenancy/v8#DB.WithTenant) to simplify this process.

We have implemented conformance tests in [pkg/drivertest/isolation.go](https://github.com/bartventer/gorm-multitenancy/blob/f525a80e8bd04e601489d3fa751345da1cc38d32/pkg/drivertest/isolation.go#L87), covering scenarios like concurrent tenant creation, schema creation, schema migration, and high-load operations. The PostgreSQL driver passes these tests ([latest job run](https://github.com/bartventer/gorm-multitenancy/actions/runs/10444151167/job/28918375537)).

I hope this information helps. Let me know if you have any other questions.
